staging/src/k8s.io/client-go/tools/cache/shared_informer.go

```diff
@@ -508,13 +518,13 @@ func determineResyncPeriod(desired, check time.Duration) time.Duration {
 
 const minimumResyncPeriod = 1 * time.Second
 
-func (s *sharedIndexInformer) AddEventHandlerWithResyncPeriod(handler ResourceEventHandler, resyncPeriod time.Duration) *ResourceEventHandlerHandle {
+func (s *sharedIndexInformer) AddEventHandlerWithResyncPeriod(handler ResourceEventHandler, resyncPeriod time.Duration) (*ResourceEventHandlerHandle, error) {
 	s.startedLock.Lock()
 	defer s.startedLock.Unlock()
 
 	if s.stopped {
 		klog.V(2).Infof("Handler %v was not added to shared informer because it has stopped already", handler)
-		return nil
+		return nil, fmt.Errorf("handler %v was not added to shared informer because it has stopped already", handler)
```

Comment:
do you need an error? if it is added successfully, it is not going to be nil. nil means there is an error.

Reply:
either an error or a bool second return value makes it clear that the function can fail

without that, we're asking for people to not handle the nil case correctly

Reply:
that's right. i think bool second return is better.

Reply:
Hmm, I think the Add and Remove methods should behave similar in case of used for a stopped handler. That is the same reason. Either both return an additional bool or an error. Just for symmetry.

In case of an error for the Add methods we also could get rid of the log message as requested for the remove method.

Reply:
*all* the currently existing calls of AddEventHandler do *not* check the return (since there is no return before this code change), when it is upgraded to the version with this code change, and there is an error, the error will be silently dropped (no log lines printed out).

Reply:
Ok, but then we should use in both cases a bool return

Reply:
I think i agree with you considering that there might be future error cases. For the Add method, since there is currently a log line, i think it is better to keep it to avoid surprises to the existing code users who does not handle the return (since there is no return before the change).

@liggitt what do you think about it?

Reply:
returning error lets you communicate details about the error, that seems reasonable to me